tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -3465,3 +3465,359 @@ def create_zeros_like_3_val():
     assert(     f3_shape() ==      create_zeros_like_3_shape()      )
     assert(isclose(     f3_val()  ,      create_zeros_like_3_val()        , rtol=RTOL, atol=ATOL))
     assert(type(f3_val())  == type(create_zeros_like_3_val().item()))
+
+def test_numpy_real_scalar(language):
+
+    @types('bool')
+    def test_bool(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('int')
+    def test_int(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('int8')
+    def test_int8(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('int16')
+    def test_int16(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('int32')
+    def test_int32(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('int64')
+    def test_int64(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('float')
+    def test_float(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('float32')
+    def test_float32(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('float64')
+    def test_float64(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('complex64')
+    def test_complex64(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('complex128')
+    def test_complex128(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    import numpy as np
+
+    integer = randint(1e6)
+    integer8 = np.int8(randint(0, 127))
+    integer16 = np.int16(randint(1e6))
+    integer32 = np.int32(randint(1e6))
+    integer64 = np.int64(randint(1e6))
+    fl = np.float(randint(1e6))
+    fl32 = np.float32(randint(1e6))
+    fl64 = np.float64(randint(1e6))
+
+    f_bl = epyccel(test_bool, language=language)
+    f_bl = epyccel(test_bool, language=language)
+    assert (f_bl(True) == test_bool(True))
+    assert (f_bl(False) == test_bool(False))
+    # ensuring that we calculate the result type correctly for numpy bloo convertions.
+    assert (type(f_bl(False)) == type(test_bool(False)))
+    assert (type(f_bl(True)) == type(test_bool(True)))
+
+    f_integer = epyccel(test_int, language=language)
+    # int8 variable does not accept negative numbers, see https://github.com/pyccel/pyccel/issues/722
+    f_integer8 = epyccel(test_int8, language=language)
+    f_integer16 = epyccel(test_int16, language=language)
+    f_integer32 = epyccel(test_int32, language=language)
+    f_integer64 = epyccel(test_int64, language=language)
+
+    assert (f_integer(integer) == test_int(integer))
+    # ensuring that we calculate the result type correctly for numpy int convertions.
+    assert (type(f_integer(integer)) == type(test_int(integer)))
```

Comment:
Could you add the type check for at least one numpy type, plus the floats and complexes please